satellite/audit/service.go

```diff
@@ -63,13 +83,33 @@ func (service *Service) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	service.log.Info("Audit cron is starting up")
 
-	return service.Loop.Run(ctx, func(ctx context.Context) error {
-		err := service.process(ctx)
-		if err != nil {
-			service.log.Error("process", zap.Error(err))
-		}
-		return nil
+	group, ctx := errgroup.WithContext(ctx)
+
+	group.Go(func() error {
+		return service.Loop.Run(ctx, func(ctx context.Context) (err error) {
+			defer mon.Task()(&ctx)(&err)
+			err = service.process(ctx)
+			if err != nil {
+				service.log.Error("process", zap.Error(err))
+			}
+			return nil
+		})
+	})
+
+	group.Go(func() error {
+		return service.ReservoirLoop.Run(ctx, func(ctx context.Context) (err error) {
+			defer mon.Task()(&ctx)(&err)
+			observer := NewObserver(service.log.Named("audit observer"), service.overlay, service.reservoirConfig)
+			err = service.MetainfoLoop.Join(ctx, observer)
```

Comment:
the reservoirs from the observer should be saved somewhere on the audit service so that they can be accessed from other areas